### PR TITLE
workflows/tests: remove unnecessary steps, cleanup.

### PR DESCRIPTION
**Pull request by MikeMcQuaid**
These are done by setup-homebrew. Also, be more consistent with other steps.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -83,7 +70,7 @@ jobs:
 
       - name: Upload logs
         if: failure()
-        uses: actions/upload-artifact@v2
+        uses: actions/upload-artifact@master
```
Comment (chenrui333) [inline .github/workflows/tests.yml:73]: unless we are looking for something unreleased, using tagged version would be safe and easy to manage for the future.

**Review by chenrui333 — DISMISSED** [review]
lgtm and only concern would be the `master` vs tagged version

File: .github/workflows/tests.yml (hunk 1 above)
Comment (MikeMcQuaid) [reply .github/workflows/tests.yml:73]: We're using `master` pretty consistently elsewhere so I think it's worth doing so here too.

Comment (chenrui333) [reply .github/workflows/tests.yml:73]: oh ok, sounds good!

**Review by dtrodrigues — COMMENTED** [review]
Looks like GitHub has changed the default branch on all their actions (at least `actions/cache` and `actions/upload-artifact`) to `main` vs `master`.

**Pull request merged**